packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeCommon.ts

```diff
@@ -1,43 +1,23 @@
-import Common from "@ethereumjs/common";
-import { bufferToHex } from "ethereumjs-util";
+import { Common } from "@ethereumjs/common";
 
-import { dateToTimestampSeconds } from "../../../util/date";
 import { LocalNodeConfig } from "../node-types";
 
-import { getCurrentTimestamp } from "./getCurrentTimestamp";
-
-export function makeCommon(
-  {
-    initialDate,
-    chainId,
-    networkId,
-    networkName,
-    blockGasLimit,
-    hardfork,
-  }: LocalNodeConfig,
-  stateTrie: any
-) {
-  const initialBlockTimestamp =
-    initialDate !== undefined
-      ? dateToTimestampSeconds(initialDate)
-      : getCurrentTimestamp();
-
-  return Common.forCustomChain(
-    "mainnet",
+export function makeCommon({
+  chainId,
+  networkId,
+  networkName,
+  hardfork,
+}: LocalNodeConfig) {
+  const common = Common.custom(
     {
       chainId,
       networkId,
       name: networkName,
-      genesis: {
-        timestamp: `0x${initialBlockTimestamp.toString(16)}`,
-        hash: "0x",
-        gasLimit: blockGasLimit,
-        difficulty: 1,
-        nonce: "0x0000000000000042",
-        extraData: "0x1234",
-        stateRoot: bufferToHex(stateTrie.root),
```

Comment:
what about the state root?

Reply:
It's used by `putGenesisBlock` to create the genesis block. The `Common` still accepts a genesis block. I think it does because then the `Blockchain` class will use it to create the genesis block on demand... but I'm not super sure.